src/rdkafka_cgrp.c

```diff
@@ -4003,7 +5147,14 @@ void rd_kafka_cgrp_metadata_update_check (rd_kafka_cgrp_t *rkcg, int do_join) {
                              "subscription updated from metadata change: "
                              "rejoining group",
                              RD_KAFKAP_STR_PR(rkcg->rkcg_group_id));
-                rd_kafka_cgrp_rejoin(rkcg);
+
+                if (rd_kafka_rebalance_protocol(rkcg->rkcg_rk) ==
+                    RD_KAFKA_REBALANCE_PROTOCOL_COOPERATIVE &&
+                    !rd_list_empty(rkcg->rkcg_subscribed_topics))
+                        /* partitions have already been revoked */
```

Comment:
Why? Where?

Reply:
incrreb changes behavior on metadata updates, and i haven't finished that bit yet. for now have added a note next to this, will address as i implement/test in depth.